### PR TITLE
Fix permission for mounted directory

### PR DESCRIPTION
**Pull request by absoludity**
Work-around for #2260 until the underlying problem is addressed (https://github.com/kubernetes/kubernetes/issues/2630 - though not 100%).

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/2261)

### CODE REVIEW

**Comment by foxish** [comment]
@absoludity, thanks for the fix. I would rather it be a separate statement explaining why it may be necessary to chmod permissions when using hostPath volumes.

**Comment by absoludity** [comment]
@foxish Let me know if that's more what you're after. Thanks

**Comment by foxish** [comment]
@kow3ns PTAL

**Comment by foxish** [comment]
@enisoc @kow3ns PTAL

File: docs/tutorials/stateful-application/basic-stateful-set.md [hunk 1]
```diff
@@ -284,6 +284,16 @@ web-0
 web-1
 ```
 
+Note, if you instead see 403 Forbidden responses for the above curl command,
+you will need to fix the permissions of the directory mounted by the `volumeMounts`
+(due to a [bug when using hostPath volumes](https://github.com/kubernetes/kubernetes/issues/2630)) with:
+
+```shell
+for i in 0 1; do kubectl exec web-$i -- sh -c 'chmod 755 /usr/share/nginx/html'; done
```
Comment (enisoc) [inline docs/tutorials/stateful-application/basic-stateful-set.md]: You shouldn't need a shell to run `chmod`. It should work as just:

```shell
for i in 0 1; do kubectl exec web-$i -- chmod 755 /usr/share/nginx/html; done
```

Did you find that the permissions on `index.html` are fine?

Comment (absoludity) [inline docs/tutorials/stateful-application/basic-stateful-set.md]: Thanks @enisoc , I removed the sh as suggested.

Yes, the perms on index.html are fine:

```
michael@hpmor2:~/dev/k8s/statefulsets$ for i in 0 1; do kubectl exec web-$i -- ls -la /usr/share/nginx/html/; done
total 8
drwxr-xr-x 2 root root   60 Jan 25 09:23 .
drwxr-xr-x 4 root root 4096 Jan 16 06:20 ..
-rw-r--r-- 1 root root    6 Jan 25 09:23 index.html
total 8
drwxr-xr-x 2 root root   60 Jan 25 09:23 .
drwxr-xr-x 4 root root 4096 Jan 16 06:20 ..
-rw-r--r-- 1 root root    6 Jan 25 09:23 index.html
```

**Comment by kow3ns** [comment]
/lgtm

---







*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes.github.io/2261#-:-KbQmB-dxwSKkJaXod5J:b-8ks5qa)*

**Pull request merged**